logstash-core/lib/logstash/util/modules_setting_array.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/util/password"
+
+module LogStash module Util class ModulesSettingArray
+  extend Forwardable
+  DELEGATED_METHODS = [].public_methods.reject{|symbol| symbol.to_s.end_with?('__')}
+
+  def_delegators :@original, *DELEGATED_METHODS
+
+  attr_reader :original
+  def initialize(value)
+    unless value.is_a?(Array)
+      raise ArgumentError.new("Module Settings must be an Array. Received: #{value.class}")
+    end
+    @original = value
+    # wrap passwords
+    @original.each do |hash|
+      hash.keys.select{|key| key.to_s.end_with?('password')}.each do |key|
+        hash[key] = LogStash::Util::Password.new(hash[key])
+      end
+    end
+  end
+
+  def __class__
```

Comment:
class is delegated by Forwardable to `@original`. this method is used in a test to verify that the class quacks like an array but one can test to see that it is not one.